### ISSUE TITLE
Child selectors are discarded by @griffel/webpack-loader

### ISSUE THREAD

**Issue by NotWoods** [microsoft/griffel#156]
After updating from @fluentui/react-components@9.0.0-rc.15 to @fluentui/react-components@9.0.1, we discovered that the newer Griffel version bundled with the package removes child selectors after webpack compilation.

In dev mode, where we do not compile styles ahead of time, child selectors still work. It seems to be related to storing some of the styles in a separate object.

Example of failing code:
```ts
const menuListStyles = {
  /** Loop app style overrides for Fluent UI `<MenuDivider />` component. */
  [`& .${menuDividerClassNames.root}`]: {
    ...shorthands.margin('4px', '-8px')
  },
}

/** Loop app style overrides for @fluentui/react-menu. */
const useMenuStyles = makeStyles({
  /** Loop app style overrides for Fluent UI `<MenuPopover />` component. */
  popover: {
    boxShadow: tokens.shadow28,
    ...menuListStyles
  },
  list: menuListStyles,
  /**
   * Style overrides for Fluent UI `<MenuPopover />` component that apply acrylic background.
   */
  popoverAcrylicColorized: {
    [`& .${menuItemClassNames.root}`]: {
      backgroundColor: 'transparent' // this still works
    }
  },
});
```

**Comment by layershifter** [comment]
_Can you please clarify what version of `@griffel/webpack-loader` is used?_ I made the example with following dependencies:

```json
{
  "dependencies": {
    "@fluentui/react-components": "^9.0.1",
    "@griffel/webpack-loader": "^2.0.8"
  }
}
```

Internally both Webpack loader & runtime version use the same function [`resolveStyleRulesForSlots()`](https://github.com/microsoft/griffel/blob/f2320a481c10d4aaefd6173a8585433ace14b6f0/packages/core/src/resolveStyleRulesForSlots.ts#L11) from `@griffel/core`, it resolves input to CSS and classes.

In my sandbox you can run `styles.js` with `node src/styles.js`, it returns:

```ts
[
  {
    popover: { E5pizo: 'f1btsmd1', jeaxjv: 'f1l1of6r' },
    list: { jeaxjv: 'f1l1of6r' },
    popoverAcrylicColorized: { Bso1b62: 'fcozxal' }
  },
  {
    d: [
      '.f1btsmd1{box-shadow:var(--shadow28);}',
      '.f1l1of6r .fui-MenuDivider{color:pink;}',
      '.f1l1of6r .fui-MenuDivider{color:pink;}',
      '.fcozxal .fui-MenuItem{background-color:transparent;}'
    ]
  }
]
```

To run Webpack just do `webpack`, it will output `dist/main.js`. On the bottom of that file you can find:

![image](https://user-images.githubusercontent.com/14183168/178278663-c8a04df8-ff87-48ad-949b-883b0b700de0.png)

The output from `src/styles.js` & in `dist/main.js` is the same (Webpack loader de-duplicated CSS rules in output).

---

Sorry, but it's unclear to me what does not work... Would you mind to create a simple reproduction for this and fork mine [example](https://stackblitz.com/edit/node-4pp1us?file=webpack.config.js,src%2Findex.js,package.json,dist%2Fmain.js)? You can also share a repository with a repro on Github that I can clone.

Do you have multiple version of `@griffel/*` packages in your `yarn.lock`?

**Comment by layershifter** [comment]
I debugged the issue in a product and it was **present** there. TL;DR It was a bug in `@linaria/shaker` and it's fixed already (https://github.com/callstack/linaria/commit/38a5541d26142cafa859ceffa6922ef559c57100).

---

The minimal reproduction for this case are following styles:

```tsx
const menuListStyles = {
  [`& .${menuItemClassNames.root}`]: {
    color: 'red'
  },
  [`& .${menuItemClassNames.root}:hover`]: {
    color: 'blue'
  },
};
```

While the is nothing wrong with them, they will be transformed to following JS:

```js
	var menuListStyles = (_menuListStyles = {}, (0, _defineProperty2.default)(_menuListStyles, "& .".concat(_reactComponents.menuItemClassNames.root), {
	  color: 'red'
	}), (0, _defineProperty2.default)(_menuListStyles, "& .".concat(_reactComponents.menuItemClassNames.root, ":hover"), {
	  color: 'blue'
	}), _menuListStyles);
```

And as `@linaria/sharer` has an issue with `defineProperty` that code was shaked out false positively.

---

The fix was done in `@linaria/shaker@3.0.0-beta.18` while the product was on `3.0.0-beta.15`. I will create a PR to bump dependency requirements 🚀
